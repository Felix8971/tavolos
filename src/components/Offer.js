import React from 'react';

class Offer extends React.Component {
  constructor(props) {
    super();
    
  }
  
  render() {

    return (
      <div className='offer'>
        <div className='ss'></div>
        <div className='sss'>{this.props.x}</div>
      </div>
    )
  }
}

export default Offer;

