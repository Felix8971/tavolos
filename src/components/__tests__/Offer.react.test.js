import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Offer from '../Offer';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

const data =  {
  image:'did.png',
  alt:'Dine in the dark logo',
  title:'Dine in the Dark for 1-6 people',
  voucher:'Dine int the Dark for 1-6 people (food only)',
  price:'1,400 THB',
  id:1,
};

describe('Offer', () => {

    it('should be defined', () => {
      expect(Offer).toBeDefined();
    });

    it('should render correctly', () => {
      const tree = shallow(
        <Offer data={data} />
      );
      expect(tree).toMatchSnapshot();
    });

});