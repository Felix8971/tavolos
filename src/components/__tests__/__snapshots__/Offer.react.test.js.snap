// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Offer
    data={
      Object {
        "alt": "Dine in the dark logo",
        "id": 1,
        "image": "did.png",
        "price": "1,400 THB",
        "title": "Dine in the Dark for 1-6 people",
        "voucher": "Dine int the Dark for 1-6 people (food only)",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="image"
        >
          <img
            alt="Dine in the dark logo"
            className="vignette"
            src="./images/did.png"
          />
        </div>,
        <div
          className="text"
        >
          <h2
            className="title"
          >
            Dine in the Dark for 1-6 people
          </h2>
          <p>
            <span>
              Voucher: 
            </span>
            Dine int the Dark for 1-6 people (food only)
          </p>
          <p>
            <span>
              Price: 
            </span>
            1,400 THB
          </p>
          <div
            className="btn"
          >
            Find out More
          </div>
        </div>,
      ],
      "className": "offer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Dine in the dark logo"
            className="vignette"
            src="./images/did.png"
          />,
          "className": "image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Dine in the dark logo",
            "className": "vignette",
            "src": "./images/did.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="title"
            >
              Dine in the Dark for 1-6 people
            </h2>,
            <p>
              <span>
                Voucher: 
              </span>
              Dine int the Dark for 1-6 people (food only)
            </p>,
            <p>
              <span>
                Price: 
              </span>
              1,400 THB
            </p>,
            <div
              className="btn"
            >
              Find out More
            </div>,
          ],
          "className": "text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dine in the Dark for 1-6 people",
              "className": "title",
            },
            "ref": null,
            "rendered": "Dine in the Dark for 1-6 people",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Voucher: 
                </span>,
                "Dine int the Dark for 1-6 people (food only)",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Voucher: ",
                },
                "ref": null,
                "rendered": "Voucher: ",
                "type": "span",
              },
              "Dine int the Dark for 1-6 people (food only)",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Price: 
                </span>,
                "1,400 THB",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Price: ",
                },
                "ref": null,
                "rendered": "Price: ",
                "type": "span",
              },
              "1,400 THB",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Find out More",
              "className": "btn",
            },
            "ref": null,
            "rendered": "Find out More",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image"
          >
            <img
              alt="Dine in the dark logo"
              className="vignette"
              src="./images/did.png"
            />
          </div>,
          <div
            className="text"
          >
            <h2
              className="title"
            >
              Dine in the Dark for 1-6 people
            </h2>
            <p>
              <span>
                Voucher: 
              </span>
              Dine int the Dark for 1-6 people (food only)
            </p>
            <p>
              <span>
                Price: 
              </span>
              1,400 THB
            </p>
            <div
              className="btn"
            >
              Find out More
            </div>
          </div>,
        ],
        "className": "offer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Dine in the dark logo"
              className="vignette"
              src="./images/did.png"
            />,
            "className": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Dine in the dark logo",
              "className": "vignette",
              "src": "./images/did.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="title"
              >
                Dine in the Dark for 1-6 people
              </h2>,
              <p>
                <span>
                  Voucher: 
                </span>
                Dine int the Dark for 1-6 people (food only)
              </p>,
              <p>
                <span>
                  Price: 
                </span>
                1,400 THB
              </p>,
              <div
                className="btn"
              >
                Find out More
              </div>,
            ],
            "className": "text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dine in the Dark for 1-6 people",
                "className": "title",
              },
              "ref": null,
              "rendered": "Dine in the Dark for 1-6 people",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Voucher: 
                  </span>,
                  "Dine int the Dark for 1-6 people (food only)",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Voucher: ",
                  },
                  "ref": null,
                  "rendered": "Voucher: ",
                  "type": "span",
                },
                "Dine int the Dark for 1-6 people (food only)",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Price: 
                  </span>,
                  "1,400 THB",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Price: ",
                  },
                  "ref": null,
                  "rendered": "Price: ",
                  "type": "span",
                },
                "1,400 THB",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Find out More",
                "className": "btn",
              },
              "ref": null,
              "rendered": "Find out More",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
